File: core/src/main/java/javabot/commands/LockFactoid.java
package javabot.commands;

import com.antwerkz.maven.SPI;
import com.antwerkz.sofia.Sofia;
import javabot.Message;
import javabot.dao.FactoidDao;
import javabot.model.Factoid;

import javax.inject.Inject;

@SPI({AdminCommand.class})
public class LockFactoid extends AdminCommand {
    @Param(primary = true)
    String name;

    @Inject
    private FactoidDao dao;

    @Override
    public boolean canHandle(String message) {
        return "lock".equals(message) || "unlock".equals(message);
    }

    @Override
    public void execute(final Message event) {
        final String command = args.get(0);
        if ("lock".equals(command) || "unlock".equals(command)) {
            final Factoid factoid = dao.getFactoid(name);
            if (factoid == null) {
                getBot().postMessage(event.getChannel(), event.getUser(), Sofia.factoidUnknown(name), event.isTell());
            } else if ("lock".equals(command)) {
                factoid.setLocked(true);
                dao.save(factoid);
                getBot().postMessage(event.getChannel(), event.getUser(), name + " locked.", event.isTell());
            } else if ("unlock".equals(command)) {
                factoid.setLocked(false);
                dao.save(factoid);
                getBot().postMessage(event.getChannel(), event.getUser(), name + " unlocked.", event.isTell());
            }
        }
    }
}

File: src/main/kotlin/javabot/commands/LockFactoid.kt
package javabot.commands

import com.antwerkz.sofia.Sofia
import com.beust.jcommander.Parameter
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao
import javabot.dao.FactoidDao
import javax.inject.Inject

class LockFactoid
@Inject
constructor(bot: Javabot, adminDao: AdminDao, var factoidDao: FactoidDao) :
    AdminCommand(bot, adminDao) {

    @Parameter lateinit var args: MutableList<String>

    override fun execute(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        val factoidName = args.joinToString(" ")
        val factoid = factoidDao.getFactoid(factoidName)
        if (factoid == null) {
            responses.add(Message(event, Sofia.factoidUnknown(factoidName)))
        } else {
            factoid.locked = true
            factoidDao.save(factoid)
            responses.add(Message(event, "${factoidName} locked."))
        }
        return responses
    }
}
